File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result;
        if(root == nullptr) {
            return result;
        }
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            vector<int> temp;
            int size= q.size();
            for(int i = 0 ; i < size; i++ ){
                TreeNode* tempnode= q.front();
                q.pop();
                temp.push_back(tempnode->val);
                if(tempnode->left != nullptr) q.push(tempnode->left);
                if(tempnode->right != nullptr) q.push(tempnode->right);
            }
            int num= temp.back();
            result.push_back(num);

        }
        return result;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node, prev_max):
            count=0
            next_max= prev_max
            if node.val >= next_max:
                count+=1
                next_max=node.val
            if node.left:
                count+=dfs(node.left,next_max)
            if node.right:
                count+=dfs(node.right,next_max)
            return count
        return dfs(root, root.val)